File: CS/2160_Minimum_Sum_of_Four_Digit_Number_After_Splitting_Digits.cs
public class Solution {
    public int MinimumSum(int num) {
        int result = 0;
        
        int[] digits = new int[4];
        for(int i = 0; i < 4; i++){
            digits[i] = num%10;
            num /= 10;
        }
        int temp;
        for(int i = 0; i < 4; i++){
            for(int j = i+1; j < 4; j++){
                if(digits[i] > digits[j]){
                    temp = digits[i];
                    digits[i] = digits[j];
                    digits[j] = temp;
                }
            }
        }
        
        
        return (digits[0]+digits[1])*10 + (digits[2] + digits[3]);
    }
}

File: Cpp/2160_Minimum_Sum_of_Four_Digit_Number_After_Splitting_Digits.cpp
class Solution {
public:
    int minimumSum(int num) {
        vector<int> v;
        while(num > 0){
            v.push_back(num%10);
            num /= 10;
        }
        sort(v.begin(), v.end());
        /*
        // bubble sort
        for(int i = 0; i < v.size(); i++){
            for(int j = 0; j < v.size()-1-i; j++){
                if(v[j] > v[j+1]){
                    int temp = v[j];
                    v[j] = v[j+1];
                    v[j+1] = temp;
                }
            }
        }
        */
        return (v[0] + v[1])*10 + (v[2]+v[3]);
    }
};
